test/services/invoice_rebilling.service.test.js

```diff
@@ -0,0 +1,146 @@
+        for (const licence of licences) {
+          const transactions = await TransactionModel.query().where('licenceId', licence.id)
+          const lineDescriptions = transactions.map(transaction => transaction.lineDescription)
+          const chargeCredits = transactions.map(transaction => transaction.chargeCredit)
+
+          expect(transactions.length).to.equal(3)
+          expect(chargeCredits).to.only.contain(true)
+          expect(lineDescriptions).to.only.contain(['TRANSACTION001', 'TRANSACTION002', 'TRANSACTION003'])
+        }
+      })
+    })
+
+    describe('rebill licences', () => {
+      let licences
+
+      beforeEach(async () => {
+        licences = await LicenceModel.query().where('invoiceId', rebillInvoice.id)
+      })
+
+      it('are created for each licence of the invoice', async () => {
+        const licenceNumbers = licences.map(licence => licence.licenceNumber)
+
+        expect(licenceNumbers.length).to.equal(3)
+        expect(licenceNumbers).to.only.contain(['LICENCE001', 'LICENCE002', 'LICENCE003'])
+      })
+
+      it('are populated with the original transactions', async () => {
+        for (const licence of licences) {
+          const transactions = await TransactionModel.query().where('licenceId', licence.id)
+          const lineDescriptions = transactions.map(transaction => transaction.lineDescription)
+          const chargeCredits = transactions.map(transaction => transaction.chargeCredit)
+
+          expect(transactions.length).to.equal(3)
+          expect(chargeCredits).to.only.contain(false)
+          expect(lineDescriptions).to.only.contain(['TRANSACTION001', 'TRANSACTION002', 'TRANSACTION003'])
+        }
+      })
+    })
+  })
+
+  async function createPopulatedLicence (invoice, licenceNumber) {
+    const licence = await LicenceModel.query()
+      .insert({
+        invoiceId: invoice.id,
+        billRunId: invoice.billRunId,
+        licenceNumber
+      })
+
+    await addTransactionToLicence(invoice, licence, licenceNumber, 'TRANSACTION001')
+    await addTransactionToLicence(invoice, licence, licenceNumber, 'TRANSACTION002')
+    await addTransactionToLicence(invoice, licence, licenceNumber, 'TRANSACTION003')
+
+    return licence
+  }
+
+  async function addTransactionToLicence (invoice, licence, licenceNumber, lineDescription) {
+    await TransactionHelper.addTransaction(currentBillRun.id, {
+      invoiceId: invoice.id,
+      licenceId: licence.id,
+      chargeFinancialYear: 2020,
+      customerReference: 'CUSTOMER_REFERENCE',
+      lineAttr1: licenceNumber,
+      lineDescription
+    })
+  }
+
+  async function addRebillInvoice (billRunId, customerReference, financialYear, rebilledInvoiceId, rebilledType) {
+    return InvoiceHelper.addInvoice(
+      billRunId,
+      customerReference,
+      financialYear,
+      0, 0, 0, 0, 0, 0, 0, 0,
+      rebilledInvoiceId,
+      rebilledType)
+  }
```

Comment:
Similar to my previous comment I think these might be better moved to the helpers (though which ones will probably involve a discussion!).

Something like `createPopulatedLicence()` especially. I think it and some other test setups I've seen indicate we have a real need to create populated bill runs for testing.

Reply:
As discussed we're leaving this for now due to circular referencing issues and we'll revisit test helpers later.